src/test/create-user.test.ts

```diff
@@ -1,116 +1,114 @@
-        birthdate: new Date(birthdate).toJSON(),
-        email,
-        password,
-        salt: saltTest,
-      });
+describe('Axios => Integration Testing - Create User use-case', () => {
+  class GraphQLEndPoint {
+    static getURL(): string {
+      const host: string = Container.get(HOST);
+      const port: number = Container.get(PORT);
+      return `http://${host}:${port}/`;
     }
   }
 
-  @Service()
-  class UserDataSourceTest extends UserDataSource {
-    override async saveUser(user: CreateUserInputModel) {
-      Object.assign(user, { salt: saltTest });
-      return this.userRepository.save({ ...user });
-    }
+  type GraphRespModel = {
+    createUser: CreateUserResponseModel;
+  };
+  type AxiosResp = Pick<AxiosResponse, 'data' | 'status' | 'statusText'> & { data: { errors?: any[] } };
+
+  let graphResponse: GraphRespModel & { errors: any[] };
+  let input = (user?: Partial<CreateUserInputModel>) => {
+    return `#graphql
+    mutation {
+      createUser (input: { 
+        name: "John",
+        email: "${user?.email ?? 'jo@example.com'}",
+        birthdate: "${user?.birthdate ?? '2000-01-01'}",
+        password: "${user?.password ?? 'p@ssw0rd'}"
+      }) {
+        id
+        name
+        email
+        birthdate
+      }
+    }`;
+  };
+
+  async function requisition(queryInput: string): Promise<AxiosResp> {
+    return axios.post(GraphQLEndPoint.getURL(), { query: queryInput });
   }
 
-  before(() => {
-    createUser = new CreateUserUseCaseTest();
-    saltTest = createUser.saltTest;
-  });
+  afterEach(() => clearRepository());
+
+  it('should create a new valid user', async () => {
+    const {
+      status,
+      data: { data: graphResponse },
+    } = await requisition(input());
 
-  afterEach(async () => {
-    await Database.connection.getRepository(User).clear();
+    expect(status).to.be.eq(200);
+    expect(graphResponse.createUser)
+      .to.include({
+        name: 'John',
+        email: 'jo@example.com',
+        birthdate: new Date('2000-01-01').toISOString(),
+      })
+      .and.has.property('id')
+      .that.is.a('number');
   });
 
-  it('should create a new valid user', async () => {
-    const input = {
-      name: 'John',
-      email: 'john@example.com',
-      birthdate: '2000-01-01',
-      password: 'p@ssw0rd',
-    };
-    const savedUser = await createUser.execTest(input);
-    expect(savedUser).to.deep.includes({
-      name: 'John',
-      email: 'john@example.com',
-      birthdate: new Date(input.birthdate).toISOString(),
-      salt: saltTest,
-    });
-    expect(savedUser.email).to.be.a('string');
-    expect(savedUser.birthdate).to.be.a('string');
+  it('should throw an error if the email is not provided', async () => {
+    await requisition(input());
+    const {
+      data: { graphResponse, errors },
+    } = await requisition(input({ email: '' }));
+
+    expect(graphResponse).to.be.undefined;
+    expect(errors[0].message).to.be.eq(`Campo e-mail obrigatório.`);
```

Comment:
vc pode testar tb o tamanho do array de erros, para garantir que a sua api está jogando o numero certo de erros esperados. No caso desse teste, o esperado é ser um ne?

```suggestion
    expect(errors).to.have.lengthOf(1);
    expect(errors[0].message).to.be.eq(`Campo e-mail obrigatório.`);
```